plugins/ipam/host-local/backend/disk/lock.go

```diff
@@ -15,36 +15,47 @@
 package disk
 
 import (
+	"github.com/alexflint/go-filemutex"
 	"os"
-	"syscall"
+	"path"
 )
 
 // FileLock wraps os.File to be used as a lock using flock
 type FileLock struct {
-	f *os.File
+	f *filemutex.FileMutex
 }
 
 // NewFileLock opens file/dir at path and returns unlocked FileLock object
-func NewFileLock(path string) (*FileLock, error) {
-	f, err := os.Open(path)
+func NewFileLock(lockPath string) (*FileLock, error) {
+	fi, err := os.Stat(lockPath)
+	if err != nil {
+		return nil, err
+	}
+
+	if fi.IsDir() {
+		lockPath = path.Join(lockPath, "lock")
+	}
+
+	f, err := filemutex.New(lockPath)
 	if err != nil {
 		return nil, err
 	}
 
 	return &FileLock{f}, nil
 }
 
-// Close closes underlying file
 func (l *FileLock) Close() error {
-	return l.f.Close()
+	return nil
```

Comment:
why not close the file?

Reply:
oh, because the library doesn't expose a `Close()` method at all.  uh oh....

Reply:
Although apparently we're not using `Close()` anyhow.  When I delete the `Close()` method from the `backend.Store` interface, the program still compiles.  So I guess you could just do that.

Reply:
Nice catch - will remove, though I must say with a touch of disquiet.

Reply:
So it seems like main.go:51 creates a Store struct and not the interface and calls defer store.Close so close is called after all...

Reply:
There is an open PR on the library to add a close method, will create a PR to update this plugin once that has been merged. alexflint/go-filemutex#2

Reply:
Updated with shiny new close method!